pandas/core/arrays/masked.py

```diff
@@ -1319,7 +1319,10 @@ def max(self, *, skipna: bool = True, axis: AxisInt | None = 0, **kwargs):
         return self._wrap_reduction_result("max", result, skipna=skipna, axis=axis)
 
     def map(self, mapper, na_action: Literal["ignore"] | None = None):
-        return map_array(self.to_numpy(), mapper, na_action=na_action)
+        if self.dtype.na_value is libmissing.NA:
```

Comment:
Did this need to change or is this just a performance optimization you are trying to achieve? I think this is creating an implicit assumption between the NA sentinel being used and whether or not the data should be converted to NumPy; I am a little wary of that

Reply:
Yes, manage 2 specific cases based on `libmissing.NA` is not necessary anymore ! :-) It was my 1st approach when I started this PR to avoid to cast data including `pd.NA` 
Thanks !